File: array/ts/minimum_absolute_difference.ts
function minimumAbsDifference(arr: number[]): number[][] {
    const res: number[][] = []
    if (arr.length === 0) {
        return res
    }

    // 数字必须传比较器，默认 sort 是按字典序排序
    arr.sort((a, b) => a - b)
    let min = arr[1] - arr[0]

    for (let i = 1; i < arr.length; i++) {
        const item = [arr[i - 1], arr[i]]

        if (arr[i] - arr[i - 1] === min) {
            res.push(item)
        }

        if (arr[i] - arr[i - 1] < min) {
            min = arr[i] - arr[i - 1]
            res.length = 0 // 相当于 Go 的 res[:0]
            res.push(item)
        }
    }
    return res
}

File: array/python/minimum_absolute_difference.py
from typing import List

class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        res = []
        if not arr:
            return res

        arr.sort()
        min_diff = arr[1] - arr[0]

        for i in range(1, len(arr)):
            diff = arr[i] - arr[i - 1]
            item = [arr[i - 1], arr[i]]

            if diff == min_diff:
                res.append(item)

            if diff < min_diff:
                min_diff = diff
                res.clear()
                res.append(item)

        return res
